### PR TITLE
[kube-prometheus-stack]update kubernetes-system-kube-proxy.yaml

### PR DESCRIPTION
**Pull request by jjmengze**
#### What this PR does / why we need it:

 it seems to me that if someone doesn't create the service monitor of kubeProxy , prometheus will strike alertmanager 

#### Which issue this PR fixes

fixes [kube-prometheus-stack] KubeProxyDown rule can't be disabled #1618

#### Special notes for your reviewer:

none 

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by skarj** [comment]
@jjmengze you forgot to bump Chart Version

File: charts/kube-prometheus-stack/templates/prometheus/rules-1.14/kubernetes-system-kube-proxy.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ Do not change in-place! In order to change this file first read following link:
 https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack/hack
 */ -}}
 {{- $kubeTargetVersion := default .Capabilities.KubeVersion.GitVersion .Values.kubeTargetVersionOverride }}
-{{- if and (semverCompare ">=1.14.0-0" $kubeTargetVersion) (semverCompare "<9.9.9-9" $kubeTargetVersion) .Values.defaultRules.create }}
+{{- if and (semverCompare ">=1.14.0-0" $kubeTargetVersion) (semverCompare "<9.9.9-9" $kubeTargetVersion)  .Values.defaultRules.create .Values.kubeProxy.enabled .Values.defaultRules.rules.kubeProxy}}
```
Comment (mrueg) [inline charts/kube-prometheus-stack/templates/prometheus/rules-1.14/kubernetes-system-kube-proxy.yaml:7]: This file is generated, you'd need to change the python script in the hack folder that syncs the rules.

Comment (jjmengze) [reply charts/kube-prometheus-stack/templates/prometheus/rules-1.14/kubernetes-system-kube-proxy.yaml:7]: sincere thanks

File: charts/kube-prometheus-stack/Chart.yaml [hunk 2]
```diff
@@ -17,7 +17,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 25.2.0
+version: 25.1.2
```
Comment (monotek) [inline charts/kube-prometheus-stack/Chart.yaml:20]: New feature should raise minor version.

```suggestion
version: 25.3.0
```

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/kube-prometheus-stack/Chart.yaml (hunk 2 above)
Comment (jjmengze) [reply charts/kube-prometheus-stack/Chart.yaml:20]: truly appreciate

**Review by monotek — CHANGES_REQUESTED** [review]
Coudl you please update the files via the python script. 
git diff is different if done locally.

**Comment by jjmengze** [comment]
@monotek yes sure, thanks your advices

**Review by monotek — APPROVED** [review]

**Comment by jjmengze** [comment]
rebase main branch and resolve conflict

**Review by mrueg — APPROVED** [review]

File: charts/kube-prometheus-stack/hack/sync_prometheus_rules.py [hunk 3]
```diff
@@ -112,6 +113,7 @@ def new_representer(dumper, data):
     'CoreDNSDown': '.Values.kubeDns.enabled',
     'AlertmanagerDown': '.Values.alertmanager.enabled',
     'AggregatedAPIDown': 'semverCompare ">=1.18.0-0" $kubeTargetVersion',
+    'KubeProxyDown': '.Values.kubeProxy.enabled',
```
Comment (monotek) [inline charts/kube-prometheus-stack/hack/sync_prometheus_rules.py:116]: Not sure if we need this?
The created rule looks broken if Values.kubeProxy.enabled is set to false.

Comment (jjmengze) [reply charts/kube-prometheus-stack/hack/sync_prometheus_rules.py:116]: oh sorry i did not notice that is seems empty

**Review by monotek — APPROVED** [review]

**Pull request merged**